### ISSUE TITLE
Ability to remove unnecessary Template Parts from core theme

### ISSUE THREAD

**Issue by ramizmanked** [WordPress/gutenberg#39951]
## What problem does this address?
With FSE, per best of my knowledge, we are unable to remove unnecessary/unwanted template-parts comes with default theme (i.e. TwentyTwentyTwo). 

Recently I developed a website on top of TwentyTwentyTwo theme. I wanted to customize some parts of default theme so I created a child theme of TwentyTwentyTwo and started working on it. At later stage, I noticed there are unused template-parts I no longer needed and I started looking into ways to remove those but I didn't find any solution except removing it directly from parent TwentyTwentyTwo theme.

## What is your proposed solution?
Overriding "templateParts" in child theme's theme.json should reflect on the Editor > Template Parts and it should allow modifying/deleting Template Parts.

**Comment by MatthiasReinholz** [comment]
@ramizmanked I agree, that removing template parts from the child theme could be useful. Especially, when you're building a parent theme with some kind of extensive parts library and you want to make the child theme as simple as possible to maintain for a client.

You're also mentioning "modifying" Template Parts. That's already possible. You can drop a `template-part.html` (with the similar file name used in the parent theme) to the `/parts` folder of your child theme. If you want to modify the title of the Template Part, you can additionally add it to the child theme's `theme.json` as if you wanted to register a new Template Part. As long as you're using the same name, it's not going to add a duplicate but alter the original one.

**Comment by rosa2** [comment]
Yes, very necessary because I have 10 post types and I have made templates for each one, and besides the ones from my child theme, i have extra ones when you create or edit a post. And functions to unset templates doesn't work.

```
function child_remove_page_templates( $page_templates ) {
   unset( $page_templates['page-large-header'] );
   return $page_templates;
}
add_filter( 'theme_page_templates', 'child_remove_page_templates' );
add_filter( 'theme_post_templates', 'child_remove_page_templates' );
```
The page template gets unset (I tried with the print_r), but it still shows in the edit post page.
Please resolve this. I will try by JavaScript in the edit page to hide them.

And always thanks a lot for your work :)
